plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/DeleteResourcesWizardModel.java

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.wizard.resource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.jobs.JobGroup;
+import org.jboss.tools.common.databinding.ObservablePojo;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.internal.common.core.util.KeyValueFilterFactory.KeyValueFilter;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+import org.jboss.tools.openshift.internal.ui.job.DeleteResourceJob;
+import org.jboss.tools.openshift.internal.ui.job.OpenShiftJobs;
+
+import com.openshift.restclient.NotFoundException;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.authorization.ResourceForbiddenException;
+import com.openshift.restclient.model.IPod;
+import com.openshift.restclient.model.IReplicationController;
+import com.openshift.restclient.model.IResource;
+
+/**
+ * @author Andre Dietisheim
+ */
+public class DeleteResourcesWizardModel extends ObservablePojo {
+
+	private static final String[] ALL_RESOURCE_KINDS = { ResourceKind.BUILD, ResourceKind.BUILD_CONFIG,
```

Comment:
I'm wondering if we can't get this list from the client lib ? What if we have a kubernetes deployment (k8s Deployment instead of OpenShift DeploymentConfig)

Reply:
@jeffmaury all types that are supported by the client lib are constants in the ResourceType class. But then not all of them are types that we want/can to delete (ex. Policy, PolicyByinding etc.)